src/lib/connectors/shared/remote-debugging-connector.ts

```diff
@@ -326,7 +326,12 @@ export class Connector implements IConnector {
             };
         }
 
-        if (hasAttributeWithValue(data.element, 'link', 'rel', 'icon')) {
+        const hasFaviconLinkTag = hasAttributeWithValue(data.element, 'link', 'rel', 'icon');
+        // If favicon is not present in the html, CDP will send a request to get favicon from the root directory.
+        // We should set the tag to true in this case to avoid fetching favicon twice.
+        const hasFaviconInRootDir = data.request.url.endsWith('/favicon.ico');
```

Comment:
I was talking with Qing, and you are right, if it isn't in the root, `hasFaviconLinkTag` should be true, she is going to change how to calculate `hasFaviconInRootDir`